### PR TITLE
feat: add new transactionsByWalletIds query for accounts

### PR DESCRIPTION
**Pull request by vindard**
## Description

This PR adds a new query under `Accounts` to be able to pass a list of wallet ids owned by the account to retrieve paginated transactions for.

It also adds a new `settlementCurrency` property for the graphql `Transaction` object which would likely be needed on the frontend to distinguish which wallet a transaction comes from.

### CODE REVIEW

**Review by dolcalmi — CHANGES_REQUESTED** [review]
Looks good, just a couple of comments. Also an e2e test is required.

File: src/domain/accounts/account-validator.ts [hunk 1]
```diff
@@ -2,22 +2,21 @@ import { InvalidAccountStatusError, InvalidWalletId } from "@domain/errors"
...
+): AccountValidator | ValidationError => {
+  if (account.status !== AccountStatus.Active) {
+    return new InvalidAccountStatusError()
+  }
 
-    if (accountIdFromWallet !== account.id) return new InvalidWalletId()
+  const validateWalletForAccount = (wallet: Wallet): true | ValidationError => {
+    if (wallet.accountId !== account.id)
+      return new InvalidWalletId(
+        JSON.stringify({ accountId: account.id, accountIdFromWallet: wallet.accountId }),
```
[the remark this answers is not in the document]
Comment (vindard) [reply src/domain/accounts/account-validator.ts:15]: Good point! https://github.com/GaloyMoney/galoy/pull/1325/commits/70c2fdc3922cee74fb2c30fe0fa7b6ceaadbe8e8

**Comment by UncleSamtoshi** [comment]
Looks good to me! Only thing that could be nice is some kind of default to gather all transactions for an account without having to explicitly pass in there wallet ids.

**Review by UncleSamtoshi — APPROVED** [review]

File: src/app/accounts/get-transactions-for-account.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { AccountValidator } from "@domain/accounts"
+import { RepositoryError } from "@domain/errors"
+import { WalletsRepository } from "@services/mongoose"
+
+import { Wallets } from "@app"
```
[the remark this answers is not in the document]
Comment (vindard) [reply src/app/accounts/get-transactions-for-account.ts]: Done: https://github.com/GaloyMoney/galoy/pull/1325/commits/6816f9934f874c15e5f05ceebbd86db5a48cff12

**Comment by vindard** [comment]
> Looks good to me! Only thing that could be nice is some kind of default to gather all transactions for an account without having to explicitly pass in there wallet ids.

Added 😉 https://github.com/GaloyMoney/galoy/pull/1325/commits/670d08fad62e6b05b7b2c2222ec67e04c013d9cc

**Review by dolcalmi — APPROVED** [review]

**Review by UncleSamtoshi — APPROVED** [review]

**Review by dolcalmi — APPROVED** [review]

**Pull request merged**